Source/DiscoveryV1/Discovery.swift

```diff
@@ -85,27 +85,30 @@ public class Discovery {
     }
 
     /**
-     If the response or data represents an error returned by the Discovery service,
-     then return NSError with information about the error that occured. Otherwise, return nil.
+     Use the HTTP response and data received by the Discovery service to extract
+     information about the error that occurred.
 
-     - parameter data: Raw data returned from the service that may represent an error.
-     - parameter response: the URL response returned from the service.
+     - parameter data: Raw data returned by the service that may represent an error.
+     - parameter response: the URL response returned by the service.
      */
-    func errorResponseDecoder(data: Data, response: HTTPURLResponse) -> Error {
+    func errorResponseDecoder(data: Data, response: HTTPURLResponse) -> RestError {
+
+        let statusCode = response.statusCode
+        var errorMessage: String?
+        var metadata = [String: JSON]()
```

Comment:
Why make the values for `metadata` `JSON` and not `Any`?

Reply:
I figured I would make use of RestKit's `JSON` type. Now that I think about it, `Any` is probably easier for users to deal with than a custom data type. I imagine that the few users who care to inspect `metadata` will just be printing it to the console.